src/QuadVelocityController.cpp

```diff
@@ -0,0 +1,115 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Quad Velocity controller
+//
+// Implements details involving the throttle controller
+//
+////////////////////////////////////////////////////////////////////////////
+
+#include "QuadVelocityController.hpp"
+#include "iarc7_msgs/OrientationThrottleStamped.h"
+
+using namespace Iarc7Motion;
+
+QuadVelocityController::QuadVelocityController(ros::NodeHandle& nh) :
+nh_(nh),  
+throttle_pid_(0.0, 0.0, 0.0),
+pitch_pid_(0.0, 0.0, 0.0) ,
+roll_pid_(0.0, 0.0, 0.0) ,
+yaw_pid_(0.0, 0.0, 0.0) ,
+tfBuffer_(),
+tfListener_(tfBuffer_)
+{
+
+}
+
+void QuadVelocityController::init()
+{
+    uav_control_ = nh_.advertise<iarc7_msgs::OrientationThrottleStamped>("uav_direction_command", 50);
```

Comment:
No I needed init for something else I think, then I ended up not needing it as you pointed out. Advertise does not throw exceptions, so this will be moved to the constructor.